squidpy/pl/_spatial_utils.py

```diff
@@ -0,0 +1,897 @@
+
+    if crops is not None:
+        ax.set_xlim(crops.to_tuple()[0], crops.to_tuple()[1])
+        ax.set_ylim(crops.to_tuple()[3], crops.to_tuple()[2])
+    else:
+        ax.set_xlim(cur_coords[0], cur_coords[1])
+        ax.set_ylim(cur_coords[3], cur_coords[2])
+
+    if isinstance(scalebar_dx, list) and isinstance(scalebar_units, list):
+        scalebar = ScaleBar(scalebar_dx[lib_count], units=scalebar_units[lib_count], **scalebar_kwargs)
+        ax.add_artist(scalebar)
+
+    return ax
+
+
+def _map_color_seg(
+    seg: NDArrayA,
+    cell_id: NDArrayA,
+    color_vector: NDArrayA | pd.Series[CategoricalDtype],
+    color_source_vector: pd.Series[CategoricalDtype],
+    cmap_params: CmapParams,
+    seg_erosionpx: int | None = None,
+    seg_boundaries: bool = False,
+    na_color: str | Tuple[float, ...] = (0, 0, 0, 0),
+) -> NDArrayA:
+
+    cell_id = np.array(cell_id)
+    if isinstance(na_color, tuple) and len(na_color) == 4:
+        transparency: Union[float, None] = na_color[-1]
+    else:
+        transparency = None
+
+    if is_categorical_dtype(color_vector):
+        if transparency and np.any(color_source_vector.isna()):
+            cell_id[color_source_vector.isna()] = 0
+        val_im: NDArrayA = map_array(seg, cell_id, color_vector.codes + 1)  # type: ignore
+        cols = colors.to_rgba_array(color_vector.categories)  # type: ignore
+    else:
+        val_im = map_array(seg, cell_id, cell_id)  # replace with same seg id to remove missing segs
+        cols = cmap_params.cmap(cmap_params.norm(color_vector))
+
+    if seg_erosionpx is not None:
+        val_im[val_im == erosion(val_im, square(seg_erosionpx))] = 0
+
+    seg_im: NDArrayA = label2rgb(
+        label=val_im,
+        colors=cols,
+        bg_label=0,
+        bg_color=(1, 1, 1),  # transparency doesn't really work
+    )
+    if seg_boundaries:
+        seg_bound: NDArrayA = np.clip(seg_im - find_boundaries(val_im)[:, :, None], 0, 1)
+        seg_bound = np.dstack((seg_bound, np.where(val_im > 0, 1, 0)))  # add transparency here
+        return seg_bound
+    else:
+        seg_im = np.dstack((seg_im, np.where(val_im > 0, 1, 0)))  # add transparency here
+        return seg_im
+
+
+def _prepare_args_plot(
+    adata: AnnData,
+    shape: _AvailShapes | None = None,
+    color: Sequence[str | None] | str | None = None,
+    groups: _SeqStr = None,
+    img_alpha: Optional[float] = None,
+    alpha: Optional[float] = None,
+    use_raw: bool | None = None,
+    layer: str | None = None,
+    palette: Palette_t = None,
+) -> ColorParams:
+    alpha = 1.0 if alpha is None else alpha
+    img_alpha = 1.0 if img_alpha is None else img_alpha
+
+    # make colors and groups as list
+    groups = [groups] if isinstance(groups, str) else groups
+    color = [color] if isinstance(color, str) or color is None else color
+
+    # set palette if missing
+    for c in color:
+        if c in adata.obs.columns and is_categorical_dtype(adata.obs[c]) and c is not None:
```

Comment:
I guess `c is not None` should be first in the `if` or completely removed, since 99.999% it will not be a column in `.obs`